src/issuance.rs

```diff
@@ -310,18 +333,27 @@ impl IssueBundle<Unauthorized> {
                     &mut rng,
                 );
 
+                notes.push(note);
+
                 IssueAction {
                     asset_desc,
-                    notes: vec![note],
+                    notes,
                     finalize: false,
                 }
             }
         };
 
+        let reference_notes = if first_issuance {
+            HashMap::from([(asset, reference_note)])
+        } else {
+            HashMap::new()
+        };
+
```

Comment:
Let's move this inside the upper `if first_issuance {` to have everything in one place.

Reply:
Done